### DISCUSSION TITLE
Add support for `@oneOf` input objects

### DISCUSSION THREAD

**Discussion by steve-marmalade** [mirumee/ariadne#1193] [Ideas]
It looks like the community has landed on a solution for describing a "union of inputs" in a GraphQL schema: https://github.com/graphql/graphql-spec/pull/825

I'd love to use this in my Ariadne server. Does that PR demonstrate enough momentum around the feature to consider implementing it?

**Comment by upcFrost** [comment]
Strawberry has already implemented it: https://github.com/strawberry-graphql/strawberry/pull/3429/files

**Comment by DamianCzajkowski** [comment]
Thank you for the idea! I'd be happy to implement this, and I'm adding it to my list of things to work on right now.

**Reply by upcFrost** [reply]
Can it be done with ariadne alone? From what i see it will require porting https://github.com/graphql/graphql-js/pull/3513 to the python graphql-core lib (https://github.com/graphql-python/graphql-core/issues/219), otherwise both the introspection and the validation will be broken

**Reply by plainsane** [reply]
i think these have been closed/merged now, i would really like this feature, any idea what the timeline looks like?

**Reply by fedirz** [reply]
Created this [pull request](https://github.com/mirumee/ariadne/pull/1257) bumping `graphql-core` version.
